File: frontend/demo/component/avatar/avatar-group-max-items.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/avatar-group';
import { html, LitElement } from 'lit';
import { customElement, state } from 'lit/decorators.js';
import { getPeople } from 'Frontend/demo/domain/DataService';
import type Person from 'Frontend/generated/com/vaadin/demo/domain/Person';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('avatar-group-max-items')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  @state()
  private items: Person[] = [];

  protected override async firstUpdated() {
    const { people } = await getPeople({ count: 6 });
    this.items = people;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-avatar-group
        .maxItemsVisible="${3}"
        .items="${this.items.map((person) => ({
          name: `${person.firstName} ${person.lastName}`,
        }))}"
      ></vaadin-avatar-group>
      <!-- end::snippet[] -->
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/avatar/AvatarGroupMaxItems.java
package com.vaadin.demo.component.avatar;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.demo.domain.DataService;
import com.vaadin.demo.domain.Person;
import com.vaadin.flow.component.avatar.AvatarGroup;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.router.Route;
import java.util.List;

@Route("avatar-group-max-items")
public class AvatarGroupMaxItems extends Div {

  private List<Person> people = DataService.getPeople(6);

  public AvatarGroupMaxItems() {
    // tag::snippet[]
    AvatarGroup avatarGroup = new AvatarGroup();
    avatarGroup.setMaxItemsVisible(3);

    for (Person person : people) {
      String name = person.getFirstName() + " " + person.getLastName();
      AvatarGroup.AvatarGroupItem avatar = new AvatarGroup.AvatarGroupItem(name);
      avatarGroup.add(avatar);
    }
    // end::snippet[]

    add(avatarGroup);
  }

  public static class Exporter extends DemoExporter<AvatarGroupMaxItems> { // hidden-source-line
  } // hidden-source-line
}
